### PR TITLE
Add Separable Conv1D layer

### PR DESCRIPTION
**Pull request by Shashi456**
Test and build pass.
Just a question, do we hope to eventually support NCWH formats? A lot of code would need to be modified in that case and I'm realizing that writing these layers. 
Could i open an issue for that or would you want to support all layers first before supporting multiple formats?

### CODE REVIEW

**Comment by jon-tow** [comment]
@Shashi456 A slight nitpick; could you please move the `Separable` layers below the `DepthwiseConv`s since the two are closely related and `ZeroPadding`s stands between them. 
It'd then also match the testing order.

**Comment by Shashi456** [comment]
@jon-tow alright. But would a layer reorg be appropriate in this PR?

**Comment by Shashi456** [comment]
@rxwei done. With this and transposed conv1d and 3d, we are done with basically supporting all convolutional layers :). Extensive testing and more functionality remains.

File: Tests/TensorFlowTests/LayerTests.swift [hunk 1]
```diff
@@ -210,6 +210,22 @@ final class LayerTests: XCTestCase {
         XCTAssertEqual(output, expected)
     }
 
+    func testSeparableConv1D() {
+        let depthwiseFilter =  Tensor(shape: [2, 2, 2], scalars: (0..<8).map(Float.init))
```
Comment (rxwei) [inline Tests/TensorFlowTests/LayerTests.swift]:
```suggestion
        let depthwiseFilter = Tensor(shape: [2, 2, 2], scalars: (0..<8).map(Float.init))
```

File: Tests/TensorFlowTests/LayerTests.swift [hunk 2]
```diff
@@ -210,6 +210,22 @@ final class LayerTests: XCTestCase {
...
+        let pointwiseFilter =  Tensor(shape: [1, 4, 1], scalars: (0..<4).map(Float.init))
```
Comment (rxwei) [inline Tests/TensorFlowTests/LayerTests.swift]:
```suggestion
        let pointwiseFilter = Tensor(shape: [1, 4, 1], scalars: (0..<4).map(Float.init))
```

**Comment by Shashi456** [comment]
Can someone trigger a build on this ? And sorry for the persistence but #288 and #174 can be merged.

**Pull request merged**